### PR TITLE
Enable verbose logging and kernel level logging

### PR DESCRIPTION
**Pull request by DonJayamanne**
More logging to see what's going on
Enable logging of messages within IPykernel, so we can see what's going on in there.

### CODE REVIEW

File: src/client/datascience/kernel-launcher/kernelProcess.ts [hunk 1]
```diff
@@ -437,7 +438,11 @@ export class KernelProcess implements IKernelProcess {
 
             // If we don't have a KernelDaemon here & we're not running a Python module either.
             // The kernelspec argv could be something like [python, main.py, --something, --something-else, -f,{connection_file}]
-            exeObs = executionService.execObservable(this.launchKernelSpec.argv.slice(1), {
+            const args = this.launchKernelSpec.argv.slice(1);
+            if (this.jupyterSettings.enablePythonKernelLogging) {
+                args.push('--debug');
```
Comment (DonJayamanne) [inline src/client/datascience/kernel-launcher/kernelProcess.ts:443]: Need to see what's going on inside the kernel

Comment (rchiodo) [reply src/client/datascience/kernel-launcher/kernelProcess.ts:443]: LOL. I have this as the default on my own machine.

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/kernel-launcher/kernelProcess.ts [hunk 2]
```diff
@@ -349,7 +350,6 @@ export class KernelProcess implements IKernelProcess {
             // a different process tries to open it.
             const tempFile = await this.fileSystem.createTemporaryLocalFile('.json');
             this.connectionFile = tempFile.filePath;
-            await tempFile.dispose();
```
Comment (DonJayamanne) [inline src/client/datascience/kernel-launcher/kernelProcess.ts:352]: This is wrong, we need the file to exist, the next line we attempt to write to the file.
What is happening is sometimes the file is created and sometimes its deleted after we connect.

@rchiodo (found this based on the error messages logged by the kernel after adding `--debug`)
**New changes after approval**

Comment (rchiodo) [reply src/client/datascience/kernel-launcher/kernelProcess.ts:352]: Should the dispose happen later then? Like maybe in the shutdown of the kernel?

Comment (DonJayamanne) [reply src/client/datascience/kernel-launcher/kernelProcess.ts:352]: Will do that as part of https://github.com/microsoft/vscode-jupyter/issues/8827

**Pull request merged**